### PR TITLE
feat: Add optional Parent field to Device objects.

### PR DESCRIPTION
**Pull request by eaton-coreymutter**
BREAKING CHANGE: Parent parameter added to edgex_add_device(), edgex_update_device().

Part of edgexfoundry/edgex-go#4769 implementation but does not close it on its own.

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break) Flagged - added "parent" parameter to two public functions
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) None already in repo
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) Docs (OpenAPI) to update are in edgex-go, will include with core-metadata PR

## Testing Instructions

I tested using device-random in the examples, both against a core-metadata that supports the parent field (and creating a child device), and against an unmodified core-metadata (without the parent field set in the device yaml file). The devices created succesfully and persisted successfully.

### CODE REVIEW

File: src/c/edgex-rest.c [hunk 1]
```diff
@@ -729,6 +741,14 @@ edgex_device *edgex_device_dup (const edgex_device *e)
 {
   edgex_device *result = malloc (sizeof (edgex_device));
   result->name = strdup (e->name);
+  if (e->parent)
```
Comment (FelixTing) [inline src/c/edgex-rest.c:744]: Using ternary operator here could make the code more concise
`result->parent = e->parent ? strdup(e->parent) : NULL;`

**Review by FelixTing — APPROVED** [review]
LGTM

File: src/c/edgex-rest.c [hunk 2]
```diff
@@ -729,6 +741,14 @@ edgex_device *edgex_device_dup (const edgex_device *e)
...
+  {
+    result->parent = strdup (e->parent);
+  }
+  else
+  {
+    result->parent = NULL;
+  }
```
Comment (FelixTing) [inline src/c/edgex-rest.c:751]:
```suggestion
result->parent = e->parent ? strdup(e->parent) : NULL;
```
Using ternary operator here could make the code more concise

**Review by FelixTing — CHANGES_REQUESTED** [review]

File: src/c/edgex-rest.c (hunk 2 above)
Comment (eaton-coreymutter) [reply src/c/edgex-rest.c:751]: Agreed.

**Review by FelixTing — APPROVED** [review]
LGTM

**Pull request merged**